src/librustc_codegen_llvm/context.rs

```diff
@@ -290,6 +294,17 @@ impl<'ll, 'tcx> CodegenCx<'ll, 'tcx> {
 
         let isize_ty = Type::ix_llcx(llcx, tcx.data_layout.pointer_size.bits());
 
+        // If we are putting SIR into the binary or dumping it to disk, then create a SirCx. Also
+        // skip generating SIR for build scripts (we will never trace them).
+        let sir_cx = if (tcx.sess.opts.cg.tracer.encode_sir() ||
+            tcx.sess.opts.output_types.contains_key(&OutputType::YkSir)) &&
+            tcx.crate_name(hir::def_id::LOCAL_CRATE).as_str() != "build_script_build"
```

Comment:
Is `build_script_build` guaranteed not to clash with some unfortunately named user crate?

Reply:
no, but that's the name the compiler assigns a build script, so it's kind of out of our control.